### ISSUE TITLE
chap 11, SELU 

### ISSUE THREAD

**Issue by ylqi007** [ageron/handson-ml#354]
When I tried to run the Example of SELU, I got the following error:

```
Traceback (most recent call last):
  File "/home/work/PycharmProjects/Tensorflow_Tutorials/Hands-on_Machine_Learning/chap11_example_1.py", line 225, in <module>
    hidden1 = tf.layers.dense(X, n_hidden1, activation=selu, name="hidden1")
  File "/home/.local/lib/python3.5/site-packages/tensorflow/python/layers/core.py", line 184, in dense
    return layer.apply(inputs)
  File "/home/.local/lib/python3.5/site-packages/tensorflow/python/keras/engine/base_layer.py", line 828, in apply
    return self.__call__(inputs, *args, **kwargs)
  File "/home/.local/lib/python3.5/site-packages/tensorflow/python/layers/base.py", line 364, in __call__
    outputs = super(Layer, self).__call__(inputs, *args, **kwargs)
  File "/home/.local/lib/python3.5/site-packages/tensorflow/python/keras/engine/base_layer.py", line 769, in __call__
    outputs = self.call(inputs, *args, **kwargs)
  File "/home/.local/lib/python3.5/site-packages/tensorflow/python/keras/layers/core.py", line 951, in call
    return self.activation(outputs)  # pylint: disable=not-callable
  File "/home/work/PycharmProjects/Tensorflow_Tutorials/Hands-on_Machine_Learning/chap11_example_1.py", line 185, in selu
    return scale * elu(z, alpha)
  File "/home/work/PycharmProjects/Tensorflow_Tutorials/Hands-on_Machine_Learning/chap11_example_1.py", line 161, in elu
    return np.where(z < 0, alpha * (np.exp(z) - 1), z)
AttributeError: 'Tensor' object has no attribute 'exp'
```

And the error results from the following part:

```
# ELU
def elu(z, alpha=1):
    return np.where(z < 0, alpha * (np.exp(z) - 1), z)

# SELU
def selu(z,
         scale=1.0507009873554804934193349852946,
         alpha=1.6732632423543772848170429916717):
    return scale * elu(z, alpha)
```

I found that usually the `z` of `elu(z, alpha=1` is `np.ndarray`, but sometimes `z` is a `<class 'tensorflow.python.framework.ops.Tensor'>`

@ageron Could you please give me some hint when you are convenient?

**Comment by ageron** [comment]
Hi @ylqi007 ,
Thanks for your question. The first error says that the `Tensor` object has no attribute `exp`, and since it happens on a line that only uses `exp()` in `np.exp(z)`, this seems to indicate that in your case, `np` is a `Tensor` instead of being the NumPy module.  Perhaps you assigned a `Tensor` to `np` at one point? Can you please check that? It should be the NumPy module:

```pycon
>>> np
<module 'numpy' from '[...]/python3.6/site-packages/numpy/__init__.py'>
```

If it's not, check your code, it should never modify `np`.  You can restart the Jupyter module and ensure that `np` never gets modified after `import numpy as np`.

Moreover, in the notebook, I first define `elu()` and `selu()` in NumPy, in order to plot a graph, but then I redefine `selu()` to have a TensorFlow version.  Perhaps you went through these cells, and then came back to earlier cells, which might not work since earlier cells expect the NumPy version, not the TensorFlow version.

In short: ensure that `np` is never modified after ìmport numpy as np`, and make sure you execute the notebook cells in order.

Hope this helps,
Aurélien
